Source/WebKit/UIProcess/API/Cocoa/_WKWebExtension.mm

```diff
@@ -42,34 +42,78 @@ @implementation _WKWebExtension
 
 #if ENABLE(WK_WEB_EXTENSIONS)
 
-- (instancetype)initWithAppExtensionBundle:(NSBundle *)appExtensionBundle
++ (instancetype)extensionWithAppExtensionBundle:(NSBundle *)appExtensionBundle
+{
+    NSError * __autoreleasing internalError;
+    auto result = WebKit::WebExtension::create(appExtensionBundle, &internalError);
+
+    if (internalError)
+        return nil;
+
+    return result->wrapper();
+}
+
++ (instancetype)extensionWithResourceBaseURL:(NSURL *)resourceBaseURL
+{
+    NSError * __autoreleasing internalError;
+    auto result = WebKit::WebExtension::create(resourceBaseURL, &internalError);
+
+    if (internalError)
+        return nil;
```

Comment:
Ditto here. Or are we expecting clients to call the API with the error and perform the logging themselves?

Reply:
I think the client should control logging or not for these.